### PR TITLE
8318106: Generated HTML for snippet does not always contain an id

### PR DESCRIPTION
**Pull request by pavelrappo**
Please review this jdk24 bug, which we intend to backport to jdk23 before RDP2, which is a week away. This is a simple fix, but it disturbs multiple tests.

Here's a tip for reviewing. Sometimes a snippet id in a test ends up with `()1`, other times with `()2`. This is because the same snippet may appear more than once on the same page, and we are always looking for its last appearance. For example, if a snippet resides in the first sentence of a method documentation, that snippet will appear in both the "Method Summary" and "Method Details" sections. Otherwise, it will only appear in "Method Details". Appearances are numbered starting from 1 and increment by 1 from top ("Method Summary") to bottom ("Method Details").

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8318106](https://bugs.openjdk.org/browse/JDK-8318106): Generated HTML for snippet does not always contain an id (**Bug** - P3)

### Reviewers
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20136/head:pull/20136` \
`$ git checkout pull/20136`

Update a local copy of the PR: \
`$ git checkout pull/20136` \
`$ git pull https://git.openjdk.org/jdk.git pull/20136/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20136`

View PR using the GUI difftool: \
`$ git pr show -t 20136`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20136.diff">https://git.openjdk.org/jdk/pull/20136.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20136#issuecomment-2223048018)

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlIds.java [hunk 1]
```diff
@@ -595,4 +595,33 @@ public HtmlId forHeading(CharSequence headingText, Set<String> headingIds) {
...
+            id += forPackage((PackageElement) e).name();
+        } else if (kind.isClass()) {
+            id += forClass((TypeElement) e).name();
+        } else if (kind.isExecutable()) {
+            id += forMember((ExecutableElement) e).getFirst().name();
+        } else if (kind.isField()) {
+            id += forMember((VariableElement) e).name();
+        } else {
+            // utmost unexpected, but this case shouldn't fail
+            id += "unknown-element-";
```
Comment (liach) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlIds.java:619]: Feels weird that unknown element has a `-` between name and count but the others don't.

Comment (pavelrappo) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlIds.java]: Sorry, I initially pushed an outdated commit. This is a rare situation where a force push is warranted. Also, fixed.

**Review by liach — APPROVED** [review]

**Pull request closed without merging**